### PR TITLE
ci: update containers and CI to use Ubuntu 22.04 LTS (`jammy`)

### PR DESCRIPTION
**Pull request by kwvg**
## Motivation

While some aspects of C++20 are supported by GCC 9.3 (the version shipped with `focal`, [source](https://packages.ubuntu.com/focal/g++)), P0896R4 ("The One Ranges Proposal") is not ([source](https://en.cppreference.com/w/cpp/compiler_support)), required for moving away from [dash#4622](https://github.com/dashpay/dash/pull/4622) to adopt `std::ranges` and for backporting PRs like [bitcoin#28687](https://github.com/bitcoin/bitcoin/pull/28687).

They're supported from GCC 10.1 onwards ([source](https://stackoverflow.com/questions/56118941/do-we-have-c20-ranges-library-in-gcc-9/56118997#56118997)) and the next available LTS, `jammy`, currently ships with GCC 11.2 ([source](https://packages.ubuntu.com/jammy/g++)). As we're now using Guix to build our releases, there shouldn't be any adverse effects from our containers having a higher version of `glibc`.

Additionally, upstream uses 24.04 (`noble`) for their build images ([source](https://github.com/bitcoin/bitcoin/blob/a2317e27b7fb86df4e32cd1674c06e09cb808248/ci/test/00_setup_env_native_tsan.sh#L10)) and `jammy` to determine the minimum version to CMake ([source](https://github.com/bitcoin/bitcoin/blob/a2317e27b7fb86df4e32cd1674c06e09cb808248/CMakeLists.txt#L5-L6)) (for reference, `focal` ships with CMake 3.16, [source](https://packages.ubuntu.com/focal/cmake)).

## Additional Information

* Builds will no longer run `check-symbols` (and the option to do so, `RUN_SYMBOL_TESTS` has been removed) as portable builds (builds expected for distributions) are built with Guix, which pins its version of `glibc` ([source](https://github.com/dashpay/dash/blob/396573d09cff5c65b2e3b4ba965185ac532b8b5c/contrib/guix/manifest.scm#L152), currently 2.28), runs `check-symbols` ([source](https://github.com/dashpay/dash/blob/396573d09cff5c65b2e3b4ba965185ac532b8b5c/contrib/guix/libexec/build.sh#L304-L305)), which in turn make sure that it doesn't contain symbols expected in versions higher than `glibc` 2.31 ([source](https://github.com/dashpay/dash/blob/396573d09cff5c65b2e3b4ba965185ac532b8b5c/contrib/devtools/symbol-check.py#L38-L42)).

  Upstream does not use `check-symbols` during their builds nor do we have a reason to (as we no longer using Gitian).

* CI build failures w.r.t. `pthread_yield` ([build](https://github.com/dashpay/dash/actions/runs/11645989968/job/32437272177?pr=6379#step:7:3136), [build](https://github.com/dashpay/dash/actions/runs/11645989968/job/32430165640#step:7:3157)) should be resolved by rebuilding `bdb4` (see [bitcoin#26423](https://github.com/bitcoin/bitcoin/issues/26423#issuecomment-1296271654)). This problem has only been observed on GitHub CI and locally (the latter resolved by rebuilding depends, I use a separate target triple for testing, `x86_64-jammy-linux-gnu`), GitLab CI has not observed such failures ([build](https://gitlab.com/dashpay/dash/-/jobs/8253943710)).

## Breaking changes

None expected.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests **(note: N/A)**
- [x] I have made corresponding changes to the documentation **(note: N/A)**
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
>CI build failures w.r.t. pthread_yield ...

Probably some cache issues. Completes with no failures in my repo https://github.com/UdjinM6/dash/actions/runs/11653234872

utACK 1592a0f92ec8531e1c1ff31e4ad61d12923f693a

File: ci/dash/build_src.sh [hunk 1]
```diff
@@ -63,7 +63,3 @@ fi
 if [ "$RUN_SECURITY_TESTS" = "true" ]; then
   make test-security-check
 fi
-
-if [ "$RUN_SYMBOL_TESTS" = "true" ]; then
```
Comment (knst) [inline ci/dash/build_src.sh:67]: Do we have guix builds for each PR? If we do that, we will find a potential issue during guix build in pre-release time, which can make mess and release delay.

I checked logs of some guix guild here https://github.com/dashpay/dash/actions/runs/11645989967/job/32429737929?pr=6379#step:20:1 and I don't see `check-symbols` running. Does it?

**Review by knst — APPROVED** [review]
utACK 1592a0f92ec8531e1c1ff31e4ad61d12923f693a

let's add release notes, that minimum version that we tested Dash Core is bumped from 20.04 to 22.04

**Comment by kwvg** [comment]
> let's add release notes, that minimum version that we tested Dash Core is bumped from 20.04 to 22.04

@knst, what determines if Guix-built binaries will work is the glibc version, 20.04 ships with glibc 2.31 ([source](https://packages.ubuntu.com/focal/glibc-doc)) while Guix is (as of this PR), targeting glibc 2.28 ([source](https://github.com/dashpay/dash/blob/1592a0f92ec8531e1c1ff31e4ad61d12923f693a/contrib/guix/manifest.scm#L516)). There wouldn't be a reason it wouldn't work on 20.04.

Also, in https://github.com/dashpay/dash/pull/6382, the glibc version will be bumped to 2.31 (leaving behind 18.04 LTS `bionic`) and that change does have release notes ([source](https://github.com/dashpay/dash/blob/45f3b34be7387dd07db309e8e4f095a08e09421c/doc/release-notes-6383.md)), which I think should be sufficient in indicating where compatibility stops.

**Review by PastaPastaPasta — APPROVED** [review]
utACK 1592a0f92ec8531e1c1ff31e4ad61d12923f693a

**Pull request merged**